### PR TITLE
Improve docstrings in Metadata API to be more descriptive

### LINKED ISSUE #1600: Metadata API: Improve docstrings language
Let's fix some docstrings in metadata.py: some of them document internal details or just repeat the name of the attribute/argument in a way that's not useful for reader:
* Assume reader is not going to read the source code 
* If it's not obvious, explain what argument/attribute is for -- why is it exposed in API?

I'd like to get at least this common line fixed:

> **unrecognized fields**: Dictionary of all unrecognized fields.

This does not explain what they are or why they are exposed in the API.

### PR DESCRIPTION
**Pull request by ivanayov**
This change updates some parts of the Metadata API docstrings
that did not give enough details and context

Fixes #1600

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [n/a] Tests have been added for the bug fix or new feature
- [n/a] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: tuf/api/metadata.py [hunk 1]
```diff
@@ -307,7 +307,8 @@ def sign(
         """Creates signature over ``signed`` and assigns it to ``signatures``.
 
         Args:
-            signer: A securesystemslib.signer.Signer implementation.
+            signer: Custom or securesystemslib implementation of the
+                ``securesystemslib.signer.Signer`` interface.
```
Comment (lukpueh) [inline tuf/api/metadata.py]: This change does not seem like a huge improvement wrt https://github.com/theupdateframework/python-tuf/pull/1815#pullrequestreview-875726152.

What about
```suggestion
            signer: A ``securesystemslib.signer.Signer`` object that provides a private
                key and signing implementation to generate the signature. A standard
                implementation is available in ``securesystemslib.signer.SSlibSigner``.
```

**Review by lukpueh — COMMENTED** [review]
Thanks for the updates, @ivanayov! I am unsure if the "dict -> json/dict" changes really add much clarification, but I don't object.

I do have an alternative suggestion for the `Signer` docstring though. Let me know what you think.

**Comment by ivanayov** [comment]
Thank you @lukpueh , I updated with your proposal.

**Review by lukpueh — APPROVED** [review]
Thanks!

**Pull request merged**